pandas/tests/indexes/test_category.py

```diff
@@ -125,6 +131,23 @@ def test_construction_with_dtype(self):
         result = CategoricalIndex(idx, categories=idx, ordered=True)
         tm.assert_index_equal(result, expected, exact=True)
 
+    def test_construction_with_categorical_dtype(self):
+        # construction with CategoricalDtype
+        # GH18109
+        data, cats, ordered = 'a a b b'.split(), 'c b a'.split(), True
+        dtype = CategoricalDtype(categories=cats, ordered=ordered)
+
+        result = pd.CategoricalIndex(data, dtype=dtype)
+        expected = pd.CategoricalIndex(data, categories=cats,
+                                       ordered=ordered)
+        tm.assert_index_equal(result, expected, exact=True)
+
+        # error to combine categories or ordered and dtype keywords args
+        with pytest.raises(ValueError):
```

Comment:
We generally try to avoid using `pytest.raises` because `tm.assert_raises_regex` is more effective.  What's the error message that you get when calling the constructor as such?

Reply:
``pytest.raises`` is ok, sure if you really want to check the message then use ``assert_raises_regex``, i don't think its a big deal here

Reply:
Well, that's why I'm asking.

Reply:
The message is "Cannot specify both `dtype` and `categories` or `ordered`." in both cases.

FYI, pytest can now take a parameter ``match`` which does the same thing (i.e. ``with pytest.raises(ValueError, match="Cannot specify both `dtype` and `categories` or `ordered`."
):``). This is a new feature added in pytest 3.1, so its very new. Pandas doesn't seem to have a minimum specified version of pytest, I assume its reasonable to be able to test pandas with version < 3.1 of pytest, though.

Reply:
actually we do have a min version of 3.1

Reply:
@jreback : Just for reference, where do we say that?

Reply:
Ok, then I'll just add the error message, as it's quick and also a good thing for documentation.

Where does it say that the minimum version for pytest is 3.1?

Reply:
it’s in our travis install, the requirements files and i believe the docs

Reply:
Ok, found it in ``contributing.rst``: "The earliest supported pytest version is 3.1.0.".

It would make sene to have it in ``install.rst`` also as there is a section there on running the test suite.

Reply:
the section in install should simply point to contributing 
we don’t wants things in more than one place

Reply:
@jreback : Do you think it would make sense to no longer `tm.assert_raises_regex` if we have `pytest.raises` in our toolbox now (unrelated to PR)?

Reply:
i don’t see a lot of benefit of changing
but could be persuaded